doc/source/file_formats.rst

```diff
@@ -1,23 +1,545 @@
+Martini, the process is typically called 'forward mapping'.
+
+The vermouth library currently utilizes two mapping formats. The ``.map``
+format, which was originally developed for the backward program, is
+used to describe how two :ref:`blocks <data:block>` correspond to each
+other. The second format (``.mapping``) is exclusively used in the
+context of :ref:`modifications <data:modification>` and is an extension
+to the first format.
+
+File structure (.map)
+~~~~~~~~~~~~~~~~~~~~~
+
+The file is structured into sections, each beginning with a directive
+enclosed in square brackets (``[]``).
+
+Allowed directives .map
+^^^^^^^^^^^^^^^^^^^^^^^
+The format recognizes the following directives:
+
+- ``[molecule]``
+    - This directive is immediately followed by a single line containing
+      an alphanumeric string specifying the residue name. This name
+      denotes the residue under consideration.
+    - ``mandatory``
+
+- ``[from]``
+    - The directive is followed by a single line containing an
+      alphanumeric string corresponding to name of the origin
+      (i.e. higher resolution) force field.
+    - ``mandatory``
+
+- ``[to]``
+    - The directive is followed by a single line containing an
+      alphanumeric string corresponding to the name of the target
+      (i.e. higher resolution) force field.
+    - ``mandatory``
+
+- ``[martini]``
+    - The directive is followed by any number of lines. Each line must
+      contain space separated bead names.
+    - ``mandatory``
+- ``[atoms]``
+    - This directive introduces a section that can span multiple lines.
+      Each line within this section must adhere to the following
+      format:
+
+      - An integer specifying the atom number.
+      - An alphanumeric string corresponding to an atom name in the
+        origin force field.
+      - Any number of bead names. These beads must have been previously
+        listed under the ``[martini]`` directive.
+    - ``mandatory``
+- ``[chiral]``
+    - Contains chirality specifications used for the original backwards
+      program.
+    - ``ignored``
+- ``[trans]``
+    - Contains geometry specifications used for the original backwards
+      program.
+    - ``ignored``
+- ``[out]``
+    - Contains geometry specifications used in the original backwards
+      program.
+    - ``ignored``
+
+
+Example of .map file
+^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: 
+
+    [ molecule ]
+    ALA ALA
+    [ martini ]
+    BB SC1
+    [ atoms ]
+     1     N    BB
+     2    HN    BB
+     3    CA    BB
+     5    CB    SC1
+     9     C    BB
+    10     O    BB
+
+
+File structure (.mapping)
+~~~~~~~~~~~~~~~~~~~~~~~~~
+The file is structured into sections, each beginning with a directive
+enclosed in square brackets (``[]``).
+
+Allowed directives .mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- ``[modification]``
+    - Marks the beginning of a modification block.
+      This directive does not require any following content.
+    - ``mandatory``
```

Comment:
Note that this is only a half-truth. .mapping files also work for Blocks (and parts of blocks), and can do some fancy things such as bridging residue boundaries.We can probably leave that out though.

Reply:
agreed! Let's keep that for us now